File: Android/app/src/main/java/com/example/aquacontrol/ContextProvider.java
package com.example.aquacontrol;

import android.app.Application;
import android.content.Context;

public class ContextProvider extends Application {
    private static ContextProvider instance;

    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    public static Context getContext() {
        return instance.getApplicationContext();
    }
}

File: Android/app/src/main/java/com/example/aquacontrol/ContextProvider.kt
package com.example.aquacontrol

import android.app.Application
import android.content.Context

class ContextProvider : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    companion object {
        private var instance: ContextProvider? = null

        @JvmStatic
        val context: Context
            get() = instance!!.applicationContext
    }
}
